src/queue.rs

```diff
@@ -432,9 +622,149 @@ impl<M: GuestAddressSpace> Queue<M> {
         fence(Ordering::Release);
 
         // We are guaranteed to be within the used ring, this write can't fail.
-        snapshot
-            .write_obj(self.next_used.0 as u16, used_ring.unchecked_add(2))
+        mem.write_obj(self.next_used.0, used_ring.unchecked_add(2))
             .unwrap();
+        if !self.event_idx_enabled {
+            // Ensure visibility of virtq_used.idx before sending notification to guest
+            // when the EVENT_IDX feature is disabled, otherwise used_event() will ensure
+            // visibility of virtq_used.idx.
+            fence(Ordering::Release);
```

Comment:
This is not needed; please remove. Sending a notification (interrupt) is itself a "release" operation and is ordered after previous stores. Just like `used_event()` contains a fence, whatever sends the interrupt (for example a KVM ioctl) will take care of it.

The code is correct without this fence. Note that

```rust
    fence(Ordering::Release);
    fence(Ordering::Acquire);
```

is not the same as `fence(Ordering::SeqCst);`.

Reply:
Thanks for explanation, removed.